### PR TITLE
Add 'is_extension' to validator metadata

### PR DESCRIPTION
**Pull request by diox**
Another bit needed for legacy submissions restrictions. I thought `type` was already exposed in `metadata`, but it's not, so I'm just adding this small bit.

### CODE REVIEW

File: validator/submain.py [hunk 1]
```diff
@@ -69,6 +69,9 @@ def timeout_handler(signum, frame):
             package = open(path, 'rb')
             test_package(err, package, path, expectation, for_appversions)
 
+        if err.detected_type == PACKAGE_EXTENSION:
+            err.metadata['is_extension'] = True
+
```
Comment (wagnerand) [inline validator/submain.py]: why wouldn't we set `err.metadata['is_extension'] = False` in the _else_ case?

File: validator/submain.py [hunk 2]
```diff
@@ -69,6 +69,9 @@ def timeout_handler(signum, frame):
             package = open(path, 'rb')
             test_package(err, package, path, expectation, for_appversions)
 
+        if err.detected_type == PACKAGE_EXTENSION:
+            err.metadata['is_extension'] = True
```
Comment (EnTeQuAk) [inline validator/submain.py]: I wonder if, given the name this should always be `False` or `True` but never not given. Although we do have a few other keys that behave this way I think. Still, can we make this explicitly set?

**Comment by EnTeQuAk** [comment]
And there are tests failing :)

**Comment by diox** [comment]
Interesting that test doesn't fail for me locally... sigh.

File: validator/submain.py (hunk 2 above)
Comment (diox) [inline validator/submain.py]: My reasoning was that:
1) I don't know all possible weird edge cases in validator that might cause the code to use a different path and avoid that function altogether
2) The existing code already behaves this way
3) In addons-server, where we have things coming from this validator but also from the linter, we will have to assume the key could be absent anyway

Comment (EnTeQuAk) [inline validator/submain.py]: The linter bit is a good reasoning. I don't think there's any other code-path, `submain` is pretty common to be called, imho it's always called but I could easily be wrong.
